### DISCUSSION TITLE
consider using `ReadonlyStoreApi` in `useStore` helper

### DISCUSSION THREAD

**Discussion by frolic** [pmndrs/zustand#2581] [Ideas] [answered]
`UseBoundStore` types were relaxed in https://github.com/pmndrs/zustand/pull/1589 because the internals of this method did not use any of the setters, only getters.

It would be great if the `useStore` helper types also made use of `ReadonlyStoreApi` in this way, since the setters are not used and thus do not require the a full store API, only getters.

Context: I am building a library that returns a store API with only readonly methods exposed and I would like folks to be able to pass this directly into Zustand's `useStore` React helper. Instead, I am forced to export my own helper that wraps Zustand, which pins users to a specific Zustand version or would need to do the optional peer dependency dance.

The change is quite small and should not affect downstream usage: https://github.com/holic/zustand/commit/2eeee190dc818fe2c9047f260921e6e1a83b06cb

**Comment by dai-shi** [comment, marked as answer]
Yes, that sounds reasonable. Would you like to open a PR?

**Reply by frolic** [reply]
done! https://github.com/pmndrs/zustand/pull/2586
